packages/admin-panel/src/VizBuilderApp/components/DataLibrary/component/SelectedOptionWithJsonEditor.jsx

```diff
@@ -31,105 +24,47 @@ const JsonEditorPanel = styled.div`
   }
 `;
 
-const OptionPanelWithJsonEditor = styled.div`
-  display: flex;
-  justify-content: space-between;
-  flex-direction: column;
-  align-items: flex-start;
-  height: auto;
-  flex: 1;
-  min-width: 0;
-  overflow: hidden;
-`;
-
-const DownArrowIconWrapper = styled.div`
-  display: flex;
-
-  .icon-wrapper {
-    cursor: pointer;
-  }
-  .MuiSvgIcon-root {
-    transition: transform 0.3s ease;
-    transform: rotate(${({ $expanded }) => ($expanded ? '0deg' : '-90deg')});
-  }
-`;
-
 export const SelectedOptionWithJsonEditor = ({
   option, // **************************************************
   basicOption, // Option panel configs (title, description etc)
   supportsTitleEditing,
   onRemove, // ************************************************
-  setIsDragDisabled, // Json editor configs
+  setIsDragDisabled, // JSON editor configs
   optionMetaData, //
   currentValue, //
   onInvalidChange, // *****************************************
   onChange,
 }) => {
-  const [isExpanded, setIsExpanded] = useState(false);
+  const enableDrag = () => setIsDragDisabled(true);
+  const disableDrag = () => setIsDragDisabled(false);
 
-  const handleToggleExpanded = newIsExpanded => {
-    if (!newIsExpanded) {
-      // When collapsing, any invalid state is thrown away. To tell our parents
-      // this we have to trigger an onChange event
-      onChange(option);
-    }
-    setIsExpanded(newIsExpanded);
+  const onValidationError = err => {
+    if (err.length > 0) onInvalidChange(err[0].message);
   };
 
+  const Editor = (
+    <JsonEditorPanel onMouseEnter={enableDrag} onMouseLeave={disableDrag}>
+      <JsonEditor
```

Comment:
I’d be quite keen to refactor this `JsonEditor.jsx` into a TSX component in `ui-components`, but I’m completely unfamiliar with the old class-based React components, so perhaps not the most efficient use of time. Any opinions?

Reply:
Wrapping the viz builder in a theme provider wrought havoc on the layout (presumably because of the way it’s embedded within Admin Panel), so I decided to just hard-code and leave a comment

Reply:
Wrapping the Viz Builder in a theme provider wrought havoc on the layout (presumably because of how it’s embedded within Admin panel), so I decided to just hard-code with an inline comment